tests/integration/oup/test_oup_pull_ftp.py

```diff
@@ -0,0 +1,58 @@
+import datetime
+
+import pytest
+from airflow import DAG
+from airflow.models import DagBag
+from airflow.models.dagrun import DagRun
+from airflow.utils.state import DagRunState
+from busypie import SECOND, wait
+from common.pull_ftp import migrate_from_ftp
+from oup.repository import OUPRepository
+from oup.sftp_service import OUPSFTPService
+from structlog import get_logger
+
+DAG_NAME = "oup_pull_ftp"
+
+
+@pytest.fixture
+def dag():
+    dagbag = DagBag(dag_folder="dags/", include_examples=False)
+    assert dagbag.import_errors.get(f"dags/{DAG_NAME}.py") is None
+    yield dagbag.get_dag(dag_id=DAG_NAME)
+
+
+def test_dag_loaded(dag: DAG):
+    assert dag is not None
+    assert len(dag.tasks) == 2
+
+
+def test_dag_run(dag: DAG):
+    repo = OUPRepository()
+    repo.delete_all()
+    assert len(repo.find_all()) == 0
+    id = datetime.datetime.utcnow().strftime("test_oup_pull_ftp_%Y-%m-%dT%H:%M:%S.%f")
+    dagrun = dag.create_dagrun(DagRunState.QUEUED, run_id=id)
+    wait().at_most(60, SECOND).until(lambda: __test_dagrun_state(dagrun))
+    assert len(repo.find_all()) == 3
+
+
+def test_dag_migrate_from_FTP():
+    repo = OUPRepository()
+    repo.delete_all()
+    assert len(repo.find_all()) == 0
+    migrate_from_ftp(
+        OUPSFTPService(),
+        repo,
+        get_logger().bind(class_name="test_logger"),
+        **{"params": {}},
+    )
+    print(repo.find_all())
+    assert len(repo.find_all()) == 3
+
+
+def __test_dagrun_state(dagrun: DagRun):
```

Comment:
why `__`?

Reply:
because it was an additional function that was used in another test. Renamed it